crates/core/src/gamerule.rs

```diff
@@ -0,0 +1,84 @@
+//! Data sourced from: https://minecraft.gamepedia.com/Game_rule
+
+use serde::{Deserialize, Serialize};
+
+/// All game rules.
+#[derive(Serialize, Deserialize)]
+#[serde(rename_all = "camelCase")]
+pub struct GameRules {
+    announce_advancements: bool,
+    command_block_output: bool,
+    disable_elytra_movement_check: bool,
+    disable_raids: bool,
+    do_daylight_cycle: bool,
+    do_entity_drops: bool,
+    do_fire_tick: bool,
+    do_insomnia: bool,
+    do_immediate_respawn: bool,
+    do_limited_crafting: bool,
+    do_mob_loot: bool,
+    do_mob_spawning: bool,
+    do_patrol_spawning: bool,
+    do_tile_drops: bool,
+    do_trader_spawning: bool,
+    do_weather_cycle: bool,
+    drowning_damage: bool,
+    fall_damage: bool,
+    fire_damage: bool,
+    forgive_dead_players: bool,
+    // freeze_damage: bool, UPCOMING to 1.17
```

Comment:
I'm not sure how I feel about these comments... we don't have them anywhere else.

Reply:
I have removed them.